File: code-samples/social-messaging/code-snippets/reply-message.js
var platform = require('./../quick-start.js').platform;
reply_message("65ef2ffea757720007399854")






// Next line must be at the 10th line!
platform.on(platform.events.loginSuccess, function(e){
    reply_message("A-Valid-Message-Id")
});

platform.on(platform.events.loginError, function(e){
    console.log("Unable to authenticate to platform. Check credentials.", e.message)
    process.exit(1)
});

/*
  Reply to a message.
*/
async function reply_message(messageId){
  try {
    var bodyParams = {
        inReplyToContentId: messageId,
        body: "Thank you for your message! Node JS",
        templateLanguage: "en",
        autoSubmitted: true,
        public: false,
      }
    let endpoint = "/cx/social-messaging/v1/contents"
    var resp = await platform.post(endpoint, bodyParams)
    var jsonObj = await resp.json()
    console.log(jsonObj)
  } catch (e){
    console.log("Unable to reply a message. Error message:", e.message)
  }
}

File: code-samples/social-messaging/code-snippets/reply-message.rb
require 'ringcentral'

# Reply to a message
def reply_message(messageId)
  begin
    bodyParams = {
          'inReplyToContentId': messageId,
          'body': "Thank you for your message! Ruby",
          'autoSubmitted': true,
          'public': false
      }
    endpoint = "/cx/social-messaging/v1/contents"
    resp = $platform.post(endpoint, payload: bodyParams)
    puts JSON.pretty_generate(JSON.parse(resp.body.to_json))
  rescue StandardError => e
    puts ("Unable to reply to this message. " + e.to_s)
  end
end

# Authenticate a user using a personal JWT token
def login()
  begin
    $platform.authorize( jwt: "RC_JWT" )
    reply_message("A-Valid-Message-Id")
  rescue StandardError => e
    puts ("Unable to authenticate to platform. Check credentials." + e.to_s)
  end
end
